malai-core/org.malai.ts/src/interaction/library/BoxChecked.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of Malai.
+ * Copyright (c) 2009-2018 Arnaud BLOUIN Gwendal DIDOT
+ * Malai is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later version.
+ * Malai is distributed without any warranty; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ */
+
+import {TSFSM} from "../TSFSM";
+import {TerminalState} from "../../../src-core/fsm/TerminalState";
+import {BoxCheckPressedTransition} from "../BoxCheckPressedTransition";
+import {isCheckBox} from "../Events";
+import {FSMDataHandler} from "../FSMDataHandler";
+import {TSInteraction} from "../TSInteraction";
+import {WidgetData} from "../../../src-core/interaction/WidgetData";
+
+/**
+ * Class use to implement the FSM of {@link BoxChecked}, it's a single transition FSM.
+ * @author Gwendal DIDOT
+ */
+class BoxCheckedFSM extends TSFSM<BoxCheckedHandler> {
+
+    public constructor() {
+        super();
+    }
+
+    /**
+     *  Function use to build the FSM of the interaction {@link BoxChecked}. This function check if the FSM already have other state than
+     *  initState, if not it's call {@link TSFSM.buildFSM()}, then add the different state of the FSM and finally the different transitions
+     *  of the FSM, with override of existing methods, as well as new methods, in this transitions if needed.
+     * @param {BoxCheckedHandler} dataHandler : This dataHandler is a instance of {@link BoxCheckedHandler}, it is use to change
+     * the data of the widget the interaction is bind with.
+     */
+    public buildFSM(dataHandler?: BoxCheckedHandler): void {
+        if (this.states.length > 1) {
+            return ;
+        }
+
+        super.buildFSM(dataHandler);
+        const checked: TerminalState<Event> = new TerminalState<Event>(this, "checked");
+        this.addState(checked);
+
+        new class extends BoxCheckPressedTransition {
+
+            /**
+             * Function use to check if event.target is a
```

Comment:
Same thing here, action is overridden so that its comment should be put on the root method